src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/EnvelopeEventProcessor.java

```diff
@@ -149,6 +167,18 @@ private void finaliseProcessedMessage(
         }
     }
 
+    private void deadLetterTheMessage(
+        IMessage message,
+        String reason,
+        String description
+    ) throws InterruptedException, ServiceBusException {
+        messageOperations.deadLetter(message.getLockToken(), reason, description);
+
+        log.info("Message with ID {} has been dead-lettered", message.getMessageId());
```

Comment:
can we also include zip file name and jurisdiction here ?

Reply:
This would require changes to `MessageProcessingResult` and would be beneficial indeed. But rather as a separate PR.

P.S. This would've be so much simpler and already ready to use if stream of `refactor/(\d){2}-breakdown-envelope-processor` would've gone through. Just saying

Reply:
This additional information isn't crucial here - we can get all that from logs.